.github/workflows/reusable_dockers_build.yml

```diff
@@ -0,0 +1,39 @@
+# Build and push Docker images to GHCR
+name: BuildDockers
+
+on: 
+  workflow_call:
+  workflow_dispatch:
+
+permissions:
+  packages: write
+  contents: read
+
+jobs: 
+  build-dockers:
+    runs-on: ubuntu-latest
+    strategy: 
+      matrix:
+        ubuntu-version: [20.04, 22.04, 24.04]
+    outputs: 
+      status: ${{ job.status }}
+    env:
+      IMG: ghcr.io/bb-ur/umf-ubuntu-${{ matrix.ubuntu-version }}:latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@30f019fb76bb54d03ec1e716054622be511a13b2 # v3.2.0
+        with: 
+          registry: ghcr.io
+          username: bb-ur
+          password: ${{ secrets.BB_GHCR_TOKEN }}
```

Comment:
sooo, I just realized - this job won't work on PR, as there's no access to secrets. So, if I'm changing anything in the docker there's no way it's going to pass...

please propose some solution, as this will be a blocker (GHA cache? connecting docker building with Basic/Fast jobs execution...? something else...?)